### PR TITLE
Filter out valid project URL path with invalid arguments to canonical url

### LINKED ISSUE #603: Valid project URL path with invalid arguments should show page with canonical url
In our logs we regularly see malformed project urls that are an amalgamation of our old url scheme and our new url scheme.
Example:
/projects/497?section=AboutProject&id=486

When we see a url like this, the request fails with error message
TypeError: must be str, not list

Whether such urls are mistakes or malicious, we should do our best to return a valid page showing the correct canonical url.

### PR DESCRIPTION
**Pull request by zh-zhang916**
I was exploring how our URL system worked, and I think I have got some ideas, credits to @marlonkeating.  We have two versions of the URL system, which the new one contains only the path argument URL but the old one has both. 
The view function for our new URL system is the index function in civictechprojects/views.py. Since the new URL system has only the path argument URL, I filter out all the arguments, started with the '?' mark,  at the beginning of the index function. Tested it with my local machine.

resolve #603

### CODE REVIEW

**Comment by zh-zhang916** [comment]
> I was exploring how our URL system worked, and I think I have got some ideas, credits to @marlonkeating. We have two versions of the URL system, which the new one contains only the path argument URL but the old one has both.
> The view function for our new URL system is the index function in civictechprojects/views.py. Since the new URL system has only the path argument URL, I filter out all the arguments, started with the '?' mark, at the beginning of the index function. Tested it with my local machine.
> 
> resolve #603

**Review by marlonkeating — CHANGES_REQUESTED** [review]
While this change properly redirects invalid links like /projects/497?section=AboutProject&id=486, it also redirects valid links like:
/projects/1?embedded=1
/projects?sortField=-project_date_modified&issues=education

I would put the logic for this in a separate method, and write unit tests to easily test that it satisfies all the known scenarios (see https://github.com/DemocracyLab/CivicTechExchange/blob/4b8c5038422faf2fa51ef008ea67ea9d848ed060/common/tests/test_helpers.py for testing examples)

File: common/helpers/front_end.py [hunk 1]
```diff
@@ -50,6 +50,22 @@ def get_page_path_parameters(url, page_section_generator=None):
     match = page_section_generator['regex'].search(url)
     return match.groupdict()
 
+def clean_invalid_args(url_args):
```
Comment (marlonkeating) [inline common/helpers/front_end.py:53]: A cleaner way to handle the logic in this method would be to
- Extract args dictionary
- Remove id/section from dictionary
- Reconstruct url from dictionary using args_dict_to_string

Comment (marlonkeating) [inline common/helpers/front_end.py:53]: Be sure to document the behavior of this method and what it returns

**Review by marlonkeating — CHANGES_REQUESTED** [review]

File: common/helpers/front_end.py (hunk 1 above)
Comment (zh-zhang916) [reply common/helpers/front_end.py:53]: I am not sure if it is the right way to document it, but I put down behaviour, args and return using standard docstrings.

Comment (zh-zhang916) [reply common/helpers/front_end.py:53]: Thanks for pointing out that function for me to use. I have incorporated it and write the code in a much cleaner way.

**Comment by zh-zhang916** [comment]
Tested it with my local machine. Work as expected.

File: common/tests/test_helpers.py [hunk 2]
```diff
@@ -46,6 +46,12 @@ def test_get_clean_url(self):
         expected = '&clean'
         self.assertEqual(expected, get_clean_url('&amp;clean'))
 
+    def test_clean_invalid_args(self):
```
Comment (marlonkeating) [inline common/tests/test_helpers.py:49]: Should add some test cases for urls with section and id too.

**Review by marlonkeating — CHANGES_REQUESTED** [review]

**Review by marlonkeating — APPROVED** [review]

**Pull request merged**